src/Storages/StorageMergeTree.cpp

```diff
@@ -269,6 +269,24 @@ std::optional<UInt64> StorageMergeTree::totalBytes(const Settings &) const
     return getTotalActiveSizeInBytes();
 }
 
+std::optional<UInt64> StorageMergeTree::totalParts(const Settings &) const
+{
+    return getDataPartsForInternalUsage().size();
```

Comment:
`getDataPartsForInternalUsage()` also only return active parts.

Reply:
So how can I get the total parts number? I have no idea...

Reply:
check other method above, like `getAllDataParts`.